diff -ruN u-boot-eng/arch/arm/dts/microgea-mx6ull.dts u-boot-eng-b/arch/arm/dts/microgea-mx6ull.dts
--- u-boot-eng/arch/arm/dts/microgea-mx6ull.dts	2026-06-09 08:27:19.866795416 +0200
+++ u-boot-eng-b/arch/arm/dts/microgea-mx6ull.dts	2026-06-09 14:44:23.968475279 +0200
@@ -130,7 +130,6 @@
 	pinctrl-0 = <&pinctrl_gpmi_nand_1>;
 	status = "okay";
 	nand-on-flash-bbt;
-	fsl,use-minimum-ecc;
 };
 
 &iomuxc {
diff -ruN u-boot-eng/configs/microgea_mx6ull_nand_defconfig u-boot-eng-b/configs/microgea_mx6ull_nand_defconfig
--- u-boot-eng/configs/microgea_mx6ull_nand_defconfig	2026-05-23 08:59:07.951308355 +0200
+++ u-boot-eng-b/configs/microgea_mx6ull_nand_defconfig	2026-06-09 08:35:46.834299062 +0200
@@ -59,3 +59,6 @@
 CONFIG_MXC_UART=y
 CONFIG_IMX_THERMAL=y
 CONFIG_USB=y
+CONFIG_CMD_UBI=y
+CONFIG_CMD_UBIFS=y
+CONFIG_MTDPARTS=y
diff -ruN u-boot-eng/configs/microgea_mx6ull_sd_defconfig u-boot-eng-b/configs/microgea_mx6ull_sd_defconfig
--- u-boot-eng/configs/microgea_mx6ull_sd_defconfig	2026-05-23 08:59:07.951308355 +0200
+++ u-boot-eng-b/configs/microgea_mx6ull_sd_defconfig	2026-06-09 08:35:50.703339247 +0200
@@ -59,3 +59,7 @@
 CONFIG_MXC_UART=y
 CONFIG_IMX_THERMAL=y
 CONFIG_USB=y
+CONFIG_CMD_UBI=y
+CONFIG_CMD_UBIFS=y
+CONFIG_MTDPARTS=y
+
diff -ruN u-boot-eng/include/configs/microgea-mx6ull.h u-boot-eng-b/include/configs/microgea-mx6ull.h
--- u-boot-eng/include/configs/microgea-mx6ull.h	2026-06-09 08:27:21.110811258 +0200
+++ u-boot-eng-b/include/configs/microgea-mx6ull.h	2026-06-13 06:58:54.325553599 +0200
@@ -26,7 +26,7 @@
 #else
 #define BOOT_FDT "try"
 #define BOOT_FITIMAGE "no"
-#define KERNEL_IMAGE_NAME "zImage"
+#define KERNEL_IMAGE_NAME "uImage"
 #endif
 
 #undef CONFIG_LDO_BYPASS_CHECK
@@ -48,7 +48,7 @@
 #define CONFIG_SYS_I2C_SPEED		100000
 #endif
 
-#define MFG_NAND_PARTITION " mtdparts=gpmi-nand:64m(nandboot),32m(nandkernel),16m(nanddtb),-(nandrootfs) "
+#define MFG_NAND_PARTITION "mtdparts=gpmi-nand:4m(boot),120m(rootfs.0),120m(rootfs.1),-(data)"
 
 #define CONFIG_CMD_READ
 #define CONFIG_SERIAL_TAG
@@ -56,36 +56,39 @@
 
 #if defined(CONFIG_NAND_BOOT)
 #define CFG_EXTRA_ENV_SETTINGS \
+        MFG_NAND_PARTITION "\0" \
+	"mtdids=nand0=gpmi-nand\0" \
+	"image=/boot/" KERNEL_IMAGE_NAME "\0" \
 	"splashimage=0x8c000000\0" \
 	"fdt_addr=0x83000000\0" \
 	"fdt_high=0xffffffff\0"	  \
+	"fdt_file=/boot/microgea-mx6ull-gwcv4-dts1.dtb\0" \
 	"tee_addr=0x84000000\0" \
 	"boot_fit=" BOOT_FITIMAGE "\0" \
 	"console=ttymxc0\0" \
-	"bootargs=console=ttymxc0,115200 ubi.mtd=nandrootfs "  \
-		"root=ubi0:rootfs rootfstype=ubifs "		     \
+	"rootpart=rootfs.0\0" \
+	"setargs=setenv bootargs console=ttymxc0,115200 ubi.mtd=${rootpart} ubi.mtd=data root=ubi0:${rootpart} rootfstype=ubifs " \
 		BOOTARGS_CMA_SIZE \
-		MFG_NAND_PARTITION \
+		" " MFG_NAND_PARTITION \
 		"\0" \
-	"bootcmd=nand read ${loadaddr} 0x4000000 0x2000000;"\
-		"nand read ${fdt_addr} 0x6000000 0x100000;"\
-		"if test ${tee} = yes; then " \
-			"nand read ${tee_addr} 0x6000000 0x400000;"\
-			"bootm ${tee_addr} - ${fdt_addr};" \
-		"else if test ${boot_fit} = yes; then " \
-			"bootm;" \
-		"else " \
-			"bootz ${loadaddr} - ${fdt_addr};" \
-		"fi;fi\0"
+	"loadfdt=ubi part ${rootpart};ubifsmount ubi0:${rootpart};ubifsload ${fdt_addr} ${fdt_file}\0" \
+	"loadimage=ubifsload ${loadaddr} ${image}\0" \
+	"bootcmd=run loadfdt;"\
+		"run loadimage;"\
+		"run setargs;"\
+		"bootm ${loadaddr} - ${fdt_addr};"\
+		"\0"
 
 #else
 #define CFG_EXTRA_ENV_SETTINGS \
+        MFG_NAND_PARTITION "\0" \
+	"mtdids=nand0=gpmi-nand\0" \
 	"script=boot.scr\0" \
 	"image="KERNEL_IMAGE_NAME"\0" \
 	"console=ttymxc0\0" \
 	"fdt_high=0xffffffff\0" \
 	"initrd_high=0xffffffff\0" \
-	"fdt_file=imx6ull-microgea-microdev-rev3.dtb\0" \
+	"fdt_file=microgea-mx6ull-gwcv4-dts1.dtb\0" \
 	"fdt_addr=0x83000000\0" \
 	"tee_addr=0x84000000\0" \
 	"tee_file=undefined\0" \
@@ -99,7 +102,7 @@
 	"mmcautodetect=yes\0" \
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
 		BOOTARGS_CMA_SIZE \
-		MFG_NAND_PARTITION \
+		" " MFG_NAND_PARTITION \
 		" root=${mmcroot}\0" \
 	"loadbootscript=" \
 		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
