diff -ruN git/include/configs/microgea-mx6ull.h git-b/include/configs/microgea-mx6ull.h
--- git/include/configs/microgea-mx6ull.h	2025-09-28 07:53:30.884000000 +0200
+++ git-b/include/configs/microgea-mx6ull.h	2025-10-11 07:47:12.776000000 +0200
@@ -191,4 +191,10 @@
 #define CONFIG_MODULE_FUSE
 #define CONFIG_OF_SYSTEM_SETUP
 
+
+#define CONFIG_SHA256
+#define CONFIG_AUTOBOOT_KEYED
+#define CONFIG_AUTOBOOT_ENCRYPTION
+#define CONFIG_AUTOBOOT_STOP_STR_SHA256  "0112e0bcfafcaf2762840c23b6bb449ec9859fbb0910518b13176bbb9b0cf801"
+
 #endif
