diff -ruN a/src/rpc_com.h b/src/rpc_com.h
--- a/src/rpc_com.h	2019-12-19 19:55:48.000000000 +0100
+++ b/src/rpc_com.h	2020-09-27 08:26:34.929169363 +0200
@@ -61,8 +61,7 @@
 void __xprt_set_raddr(SVCXPRT *, const struct sockaddr_storage *);
 
 
-SVCXPRT **__svc_xports;
-int __svc_maxrec;
+extern int __svc_maxrec;
 
 #ifdef __cplusplus
 }
diff -ruN a/src/svc.c b/src/svc.c
--- a/src/svc.c	2019-12-19 19:55:48.000000000 +0100
+++ b/src/svc.c	2020-09-27 08:26:59.673203033 +0200
@@ -57,6 +57,9 @@
 
 #define max(a, b) (a > b ? a : b)
 
+static SVCXPRT **__svc_xports;
+int __svc_maxrec;
+
 /*
  * The services list
  * Each entry represents a set of procedures (an rpc program).
